### PR TITLE
DOMA-3992 added resident app payments available indicator in ticket page

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**

### CODE REVIEW

File: apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx [hunk 1]
```diff
@@ -1,25 +1,97 @@
...
+}
+
+interface TicketResidentFeaturesProps {
+    ticket: Ticket
+}
+
+const TICKET_RESIDENT_FEATURES_ROW_GUTTER: [Gutter, Gutter] = [8, 0]
+
+export const TicketResidentFeatures: React.FC<TicketResidentFeaturesProps> = ({ ticket }) => {
+    const isContactHasMobileApp = !!get(ticket, 'client')
```
Comment (AntonAL) [inline apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx:84]: Are you sure that field `client` determines whether a contact has mobile app?
Maybe there is a `Ticket.isResidentTicket` field for that?

File: apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx [hunk 2]
```diff
@@ -1,25 +1,97 @@
...
     )
+}
+
+interface PaymentsAvailableIndicatorProps {
+    ticketOrganizationId: string
+}
+
+const getIsPaymentsInMobileAppAvailable = (receiptsInCurrentPeriod, receiptsInPreviousPeriod) => {
+    if (receiptsInCurrentPeriod > 0 || receiptsInPreviousPeriod > 0) {
+        if (receiptsInPreviousPeriod === 0) return true
```
Comment (akarjakin) [inline apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx:36]: this is confusing a bit

File: apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx (hunk 1 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx:84]: No, it's a `client` field
`isResidentTicket` it's about for whom the `ticket` was created - for a resident (a contact is created), or for an employee (in this case, a contact is not created). This flag can be set, but the resident may not have a mobile application.

File: apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx (hunk 2 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx:42]: wrote a comment

Comment (AntonAL) [reply apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx:42]: This comment just repeats what states in the code without getting extra information. Maybe to write something, like "User have new receipts, that are not fetched yet"

**Review by AntonAL — APPROVED** [review]

File: apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx [hunk 3]
```diff
@@ -1,25 +1,103 @@
+import { BillingReceipt } from '@app/condo/domains/billing/utils/clientSchema'
```
Comment (akarjakin) [inline apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx:1]: let's sort imports

**Review by akarjakin — APPROVED** [review]

File: apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx [hunk 4]
```diff
@@ -1,25 +1,103 @@
...
+    ticketOrganizationId: string
+}
+
+/**
+ Availability is determined by:
+ 1. at least 1 receipt for the current or previous period has been uploaded to the organization.
+ 2. If the number of receipts for the previous period is not equal to 0,
+ then the number of receipts for the current period must be greater than or equal to the number for the previous one.
+ */
+const getIsPaymentsInMobileAppAvailable = (receiptsInCurrentPeriod, receiptsInPreviousPeriod) => {
```
Comment (pahaz) [inline apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx:40]: probably it's easy to create something like `useIsReceiptsAvailable` which should include `BillingReceipt.useObject( ... )`

File: apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx [hunk 5]
```diff
@@ -1,25 +1,103 @@
...
+
+const PaymentsAvailableIndicator: React.FC<PaymentsAvailableIndicatorProps> = ({ ticketOrganizationId }) => {
+    const intl = useIntl()
+    const PaymentsAvailableMessage = intl.formatMessage({ id: 'pages.condo.ticket.PaymentsAvailable' })
+    const PaymentsNotAvailableMessage = intl.formatMessage({ id: 'pages.condo.ticket.PaymentsNotAvailable' })
+
+    const currentPeriod = dayjs().startOf('month').format('YYYY-MM-DD')
+    const previousPeriod = dayjs(currentPeriod).subtract(1, 'month').format('YYYY-MM-DD')
+
+    const { count: receiptsInCurrentPeriod } = BillingReceipt.useObjects({
```
[the remark this answers is not in the document]
Comment (pahaz) [reply apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx:60]: Or use count

File: apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx (hunk 4 above)
Comment (pahaz) [reply apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx:40]: and it's not enough to just have receipts. You also need a acquiring context

File: apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (nomerdvadcatpyat) [reply apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx:60]: already used count

File: apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx [hunk 6]
```diff
@@ -1,25 +1,103 @@
...
+        where: {
+            context: { organization: { id: ticketOrganizationId } },
+            period: currentPeriod,
+        },
+    })
+
+    const { count: receiptsInPreviousPeriod } = BillingReceipt.useObjects({
+        where: {
+            context: { organization: { id: ticketOrganizationId } },
+            period: previousPeriod,
```
[the remark this answers is not in the document]
Comment (nomerdvadcatpyat) [reply apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx:70]: no, we need to compare count of receipts in previous and current periods

File: apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx (hunk 4 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/ticket/components/TicketId/TicketResidentFeatures.tsx:40]: there is a condition in the receipts request `context: { organization: { id: ticketOrganizationId } }`
if there is no context, then receipts are not fetched

**Pull request merged**